registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java

```diff
@@ -532,6 +543,49 @@ private IdResponseDTO sendIdRepoWithUin(String regId, String process, JSONObject
 		return result;
 
 	}
+	
+	private void idResponseProcessUtility(IdResponseDTO idResponseDTO, InternalRegistrationStatusDto registrationStatusDto,
+			String registrationId, boolean isUinAlreadyPresent, boolean isTransactionSuccessful,
+			UinGenResponseDto uinResponseDto, MessageDTO object, LogDescription description) throws Exception {
+
+		if (isIdResponseNotNull(idResponseDTO)) {
```

Comment:
Since this method is called multiple times, this condition and handling is also done multiple times. Ideally I would like to keep this check outside this method and have 2 separate methods for handling the cases.

Reply:
Again lets do this in the develop branch changes

Reply:
implemented